src/sentry/spans/consumers/process/factory.py

```diff
@@ -119,25 +156,28 @@ def _produce_segment(message: Message[ProduceSegmentContext | None]):
             if len(keys) > 0:
                 payload_context["sample_key"] = keys[0]
 
-            total_spans_read = 0
             # With pipelining, redis server is forced to queue replies using
             # up memory, so batching the keys we fetch.
             with txn.start_child(op="process", description="produce_fetched_segments"):
                 for i in range(0, len(keys), BATCH_SIZE):
                     segments = client.read_and_expire_many_segments(keys[i : i + BATCH_SIZE])
 
                     for segment in segments:
-                        total_spans_read += len(segment)
-                        produce_segment_to_kafka(segment)
+                        if not len(segment):
```

Comment:
`if not segment` to check if the string is empty.
```
For sequences, (strings, lists, tuples), use the fact that empty sequences are false:

# Correct:
if not seq:
if seq:

# Wrong:
if len(seq):
if not len(seq):
```
https://peps.python.org/pep-0008/#programming-recommendations